linera-base/src/identifiers.rs

```diff
@@ -178,6 +178,27 @@ impl<A> Debug for BytecodeId<A> {
     }
 }
 
+impl<A> Display for BytecodeId<A> {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "BytecodeId {{ message_id: {} }}", self.message_id)
+    }
+}
+
+impl<A> Serialize for BytecodeId<A> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        if serializer.is_human_readable() {
+            serializer.serialize_str(&self.to_string())
+        } else {
+            let mut state = serializer.serialize_struct("BytecodeId", 1)?;
+            state.serialize_field("id", &self.message_id)?;
+            state.end()
```

Comment:
Good idea!
But I don't know how to get the BCS bytes in the `is_human_readable` case with this approach.

Reply:
why not `bcs::to_bytes(&self.message_id)` ?